docs/rules/no-underscore-dangle.md

```diff
@@ -42,6 +42,7 @@ This rule has an object option:
 * `"allow"` allows specified identifiers to have dangling underscores
 * `"allowAfterThis": false` (default) disallows dangling underscores in members of the `this` object
 * `"allowAfterSuper": false` (default) disallows dangling underscores in members of the `super` object
+* `"allowAfterThisConstructor": false` (default) disallows dangling underscores in members of the `this.constructor` object
```

Comment:
Enabling this check by default could cause new errors, which means this would be a breaking change if merged as-is. Since we're likely to release a new major version in the next few months, I'd just as soon leave this and merge it then. The other option would be to change this to `true` so it doesn't report any new errors by default and then update the default in the next major. Any preferences from anyone else?

Reply:
I don't believe this could possibly cause a new error to be reported. `this.constructor._foo` is already reported as an error as-is. Having this option added and default to `false` does not change that behavior of existing configs at all. Defaulting it to true would silence errors that users might not want silenced.

Reply:
Ah, I see! This is adding a special case for something that was previously handled by the general `allow` option before. You're right, this is non-breaking as-is.

Reply:
On second thought... what happens if someone had `allow: true`? In the default case, nothing changes, but if someone's configured `allow` differently, then this would remove `this.constructor._foo` from the set previously allowed by `allow: true`, wouldn't it?

Reply:
`allow` doesn't take `true` as a value. It accepts an array of identifiers which should be allowed. Like `allowAfterThis`, `allowAfterThisConstructor` is simply for relaxing the rule in a general case and would never subtract what is in `allow`.

For example:

```js
/*eslint no-underscore-dangle: ["error", { "allow": ["foo_", "_bar"], "allowAfterThisContructor": false }] */

var foo = this.constructor.foo_;
this.constructor._bar();
```

Neither of those lines would produce an error, because even though `allowAfterThisContructor` is false, both of those identifiers are in the `allow` array.

Reply:
Got it, thank you for the patient explanation. I'm on board with this :+1: